### PR TITLE
feat(remove-experiment-ui): Feature to delete archived experiments in UI

### PR DESCRIPTION
**Pull request by jatinriverlane**
# Changes
* Added delete experiment button conditionally rendered for archived experiments.
* Added deletion confirmation modal to ask user for confirmation of delete operation.

### CODE REVIEW

File: aqueductcore/frontend/src/API/graphql/mutations/Experiment/removeExperiment.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        removeExperiment(experimentRemoveInput: { experimentId: $experimentId})
+    }
+`;
+
+export function useRemoveExperiment() {
+    const [mutate, { loading, data, error }] = useMutation<{
+      removeExperiment: REMOVE_EXPERIMENT_TYPE;
+    }>(REMOVE_EXPERIMENT, {
+      onError(error) {
+        console.log("Remove experiment failed", error);
```
Comment (samiralavi) [inline aqueductcore/frontend/src/API/graphql/mutations/Experiment/removeExperiment.tsx:16]: I don't remember if we have ever used console.log in the frontend. Please check with @safeamiiir about how it should be used.

File: aqueductcore/frontend/src/components/organisms/ExperimentDetails/index.tsx [hunk 2]
```diff
@@ -240,6 +267,27 @@ function ExperimentDetails({ experimentDetails }: ExperimentDetailsProps) {
...
   };
 
+  const handleDeleteExperiment = () => {
+    if (!removeExperimentLoading) {
+      mutateRemoveExperiment({
+        variables: {
+          experimentId: experimentDetails.id
+        },
+        onError() {
+          toast.error("Failed To Delete experiment", {
```
Comment (samiralavi) [inline aqueductcore/frontend/src/components/organisms/ExperimentDetails/index.tsx:277]: Isn't it better to show the returned error message from GraphQL? It helps the user to understand the underlying issue and is also good for debugging.

**Review by samiralavi — COMMENTED** [review]
Thanks @jatinriverlane, I like to see it visually at some point. Generally, it looks good, please check with @safeamiiir my comments

**Review by samiralavi — APPROVED** [review]

File: aqueductcore/frontend/src/API/graphql/mutations/Experiment/removeExperiment.tsx (hunk 1 above)
Comment (jatinriverlane) [reply aqueductcore/frontend/src/API/graphql/mutations/Experiment/removeExperiment.tsx:16]: Sorry, missed this. There should be no console log command.

File: aqueductcore/frontend/src/components/organisms/ExperimentDetails/index.tsx [hunk 3]
```diff
@@ -375,6 +423,46 @@ function ExperimentDetails({ experimentDetails }: ExperimentDetailsProps) {
                 </BorderedButtonWithIcon>
               )}
             </Grid>
+            {isArchived(experimentDetails.tags) && <Grid item>
```
Comment (safeamiiir) [inline aqueductcore/frontend/src/components/organisms/ExperimentDetails/index.tsx:426]: We also need to check if delete is in the user's available scopes.
have a look at this file `aqueductcore/frontend/src/helper/auth/userScope.tsx`
we need to add another function with the name `isUserAbleToDeleteExperiment` and so on...

**Review by safeamiiir — CHANGES_REQUESTED** [review]
Also let's add tests, I've already added tests for experiments in my previous PRs, you can have a look and reuse some parts.

**Pull request merged**